internal/github/user.go

```diff
@@ -28,17 +27,16 @@ func NewUserHelper(gc *github.Client, repoName *RepoName) UserHelper {
 	}
 }
 
-var ErrUnexpectedReply = errors.New("the users search result are incoclusive")
+var ErrUnexpectedReply = errors.New("the number of found commits isn't exactly 1")
 
-func (uh *UserHelperImpl) GetUser(ctx context.Context, commit *object.Commit) (*github.User, error) {
+func (uh *UserHelperImpl) GetUser(ctx context.Context, sha string) (*github.User, error) {
 
-	email := commit.Author.Email
-	userSearchRes, _, err := uh.gc.Search.Users(ctx, email, nil)
+	commitSearchRes, _, err := uh.gc.Search.Commits(ctx, sha, nil)
```

Comment:
We should restrict the query to the upstream repo only - [API docs](https://docs.github.com/en/rest/search?apiVersion=2022-11-28#search-commits)

Reply:
Better docs: https://docs.github.com/en/search-github/searching-on-github/searching-commits#search-by-hash
```suggestion
	q := fmt.Sprintf("hash:%s repo:%s/%s", sha, org, repo)
	commitSearchRes, _, err := uh.gc.Search.Commits(ctx, q, nil)
```

Reply:
You are definitely right. Will do.

Reply:
Fixed.